src/main/scala/com/srcd/swivel/SparkPrepDriver.scala

```diff
@@ -0,0 +1,415 @@
+    .reduceByKeyLocally(_+_)
+    .toSeq.sortBy(_._1)
+    .foreach { case ((i, j), w) =>
+      println(s"$i $j $w")
+    }
+
+  /**
+    * Given the set of co-occurrence in a single shard, converts them to tf.Example ProtoBuf format
+    * Uses https://github.com/tensorflow/ecosystem/tree/master/hadoop#use-with-spark
+    *
+    * @param index shard number
+    * @param partition Iterator over single shard
+    * @return
+    */
+  def convertToProtobuf(index: Int, partition: Iterator[((Int, Int), Double)], numShards: Int, shardSize: Int) = {
+    val intListRow = Int64List.newBuilder()
+    val intListCol = Int64List.newBuilder()
+
+    //invers of ShardPartitioner.getPartition()
+    val row_shard = index % numShards
+    val col_shard = index / numShards
+    for (i <- 0 to shardSize - 1) {
+      intListRow.addValue(row_shard + i * numShards)
+      intListCol.addValue(col_shard + i * numShards)
+    }
+
+    val intListSparseRow = Int64List.newBuilder()
+    val intListSparseCol = Int64List.newBuilder()
+    val floatListWeight = FloatList.newBuilder()
+    partition.foreach { case ((i, j), w) =>
+      intListSparseRow.addValue(i / numShards)
+      intListSparseCol.addValue(j / numShards)
+      floatListWeight.addValue(w.toFloat) //TODO(bzz): check how we loose accuracy here
+    }
+    val features = Features.newBuilder()
+      .putFeature("global_row", Feature.newBuilder().setInt64List(intListRow.build()).build())
+      .putFeature("global_col", Feature.newBuilder().setInt64List(intListCol.build()).build())
+      .putFeature("sparse_local_row", Feature.newBuilder().setInt64List(intListSparseRow.build()).build())
+      .putFeature("sparse_local_col", Feature.newBuilder().setInt64List(intListSparseCol.build()).build())
+      .putFeature("sparse_value", Feature.newBuilder().setFloatList(floatListWeight.build()).build())
+      .build()
+
+    val example = Example.newBuilder()
+      .setFeatures(features)
+      .build()
+
+    Seq((new BytesWritable(example.toByteArray/*.toString.getBytes*/), NullWritable.get())).toIterator
+  }
+
+}
+
+
+/**
+  * Custom Partitioner to split co-occurrence matrix to smaller shards
+  *
+  * Number of partitions is square the number of shards.
+  * @param numShards number of shards (per dimention)
+  */
+class ShardPartitioner(numShards: Int) extends org.apache.spark.Partitioner {
+  require(numShards >= 0, s"Number of partitions ($numShards) cannot be negative.")
+
+  def numPartitions: Int = numShards * numShards
+
+  def getPartition(key: Any): Int = key match {
+    case null => 0
+    case (i, j) => (i.asInstanceOf[Int]%numShards) * numShards + j.asInstanceOf[Int]%numShards
+  }
+
+  override def equals(other: Any): Boolean = other match {
+    case o: ShardPartitioner => o.numPartitions == numPartitions
+    case _ => false
+  }
+
+  override def hashCode: Int = numPartitions
+}
+
+
+
+class Cli(arguments: Seq[String]) extends ScallopConf(arguments) {
```

Comment:
Sorry, we strive for following good practices and idiomatic code from scratch, specially when it is easy to do.

In Scala the idiomatic approach is keeping one class (and/or object) per file, with the name of the file being the name of the class. With the exception of groups of closely related classes that are in the same file with a name starting with lowercase (e.g. one sealed treat `Operation` and two case classes `Add` and `Sub`, file `operations.scala`).

It is easier for later contributors to follow conventions if these conventions are already followed in previous code base.